### ISSUE TITLE
Memory leak when using TypeORM data source with Jest

### ISSUE THREAD

**Issue by rconroy293** [typeorm/typeorm#9540]
## Issue Description

While upgrading from 0.2.x to 0.3.x, I'm running into what seems like a memory leak when using a test data source in my Jest test suites.

### Expected Behavior

I would expect to be able to create a real `DataSource` for use in tests and to be able to destroy that at the conclusion of each test suite.

### Actual Behavior

Heap space grows with each test.

### Steps to Reproduce

I've simplified it as best as possible down to the following:

```typescript
import { DataSource } from 'typeorm';

const testDataSource = new DataSource({
  type: 'sqlite',
  database: ':memory:',
  dropSchema: true,
  entities: [],
  synchronize: true,
});

beforeAll(async () => {
  await testDataSource.initialize();
});

afterAll(async () => {
  await testDataSource.dropDatabase();
  await testDataSource.destroy();
});

describe('example', () => {
  it('should add some things', async () => {
    expect(1 + 1).toEqual(2);
  });
});
```

If I add 50 copies of this test file and track heap usage with the following, I see the heap size increasing for about 15 tests before running into an OOM.

```
NODE_OPTIONS=--max-old-space-size=512 node --expose-gc ./node_modules/.bin/jest --runInBand --logHeapUsage
```

Example output:

```
% NODE_OPTIONS=--max-old-space-size=512 node --expose-gc ./node_modules/.bin/jest --runInBand --logHeapUsage

 PASS  test/typeorm/test-leak-50.test.ts (171 MB heap size)
 PASS  test/typeorm/test-leak-30.test.ts (201 MB heap size)
 PASS  test/typeorm/test-leak-13.test.ts (225 MB heap size)
 PASS  test/typeorm/test-leak-27.test.ts (248 MB heap size)
 PASS  test/typeorm/test-leak-29.test.ts (272 MB heap size)
 PASS  test/typeorm/test-leak-38.test.ts (296 MB heap size)
 PASS  test/typeorm/test-leak-49.test.ts (319 MB heap size)
 PASS  test/typeorm/test-leak-31.test.ts (343 MB heap size)
 PASS  test/typeorm/test-leak-28.test.ts (367 MB heap size)
 PASS  test/typeorm/test-leak-48.test.ts (390 MB heap size)
 PASS  test/typeorm/test-leak-47.test.ts (414 MB heap size)
 PASS  test/typeorm/test-leak-42.test.ts (437 MB heap size)
 PASS  test/typeorm/test-leak-26.test.ts (460 MB heap size)
 PASS  test/typeorm/test-leak-12.test.ts (430 MB heap size)
 PASS  test/typeorm/test-leak-43.test.ts (449 MB heap size)
 PASS  test/typeorm/test-leak-46.test.ts (467 MB heap size)

 RUNS  test/typeorm/test-leak-41.test.ts

<--- Last few GCs --->

[61256:0x7f7cd124d000]    14700 ms: Scavenge 486.2 (521.1) -> 485.8 (522.1) MB, 2.0 / 0.0 ms  (average mu = 0.574, current mu = 0.477) allocation failure;
[61256:0x7f7cd124d000]    14717 ms: Scavenge 487.4 (522.1) -> 486.9 (526.9) MB, 2.7 / 0.0 ms  (average mu = 0.574, current mu = 0.477) allocation failure;
[61256:0x7f7cd124d000]    14837 ms: Mark-sweep 490.7 (527.1) -> 488.2 (529.9) MB, 88.6 / 0.1 ms  (average mu = 0.546, current mu = 0.514) allocation failure; scavenge might not succeed


<--- JS stacktrace --->

FATAL ERROR: Reached heap limit Allocation failed - JavaScript heap out of memory
```

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    | macOS Monterey          |
| Node.js version     | 18.10.0  |
| Typescript version  | 4.4.4  |
| TypeORM version     | 0.3.10  |
| Jest version     | 29.1.2  |
| SQLite3 version     | 5.1.2  |

### Additional Context

N/A

### Relevant Database Driver(s)

| DB Type           | Reproducible  |
|-------------------| ---           |
| `aurora-mysql`    | no            |
| `aurora-postgres` | no            |
| `better-sqlite3`  | no            |
| `cockroachdb`     | no            |
| `cordova`         | no            |
| `expo`            | no            |
| `mongodb`         | no            |
| `mysql`           | no            |
| `nativescript`    | no            |
| `oracle`          | no            |
| `postgres`        | no            |
| `react-native`    | no            |
| `sap`             | no            |
| `spanner`         | no            |
| `sqlite`          | yes            |
| `sqlite-abstract` | no            |
| `sqljs`           | no            |
| `sqlserver`       | no            |

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✖️ Yes, I have the time, and I know how to start.
 - ✅ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✖️ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by rconroy293** [comment]
It seems as though this is related to the following known memory leak with Jest: https://github.com/facebook/jest/issues/11956

Utilizing the `workerIdleMemoryLimit` configuration allowed our tests to pass.

**Comment by dawidstezycki** [comment]
The jest issue linked has apparently been resolved in Node 21.1.0 but the problem is still there for me.
